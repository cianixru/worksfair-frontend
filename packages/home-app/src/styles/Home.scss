
.main-search-container {
    display: block;
    width: 100%;
    position: relative;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: 50%;
    height: 620px;
    z-index: 9
}

.main-search-container:before {
    content: "";
    position: absolute;
    height: 100%;
    width: 100%;
    display: block;
    background: linear-gradient(to right, rgba(255, 255, 255, .99) 20%, rgba(255, 255, 255, .7) 70%, transparent 95%)
}

.main-search-container h2 {
    font-size: 42px
}

.main-search-container h4 {
    font-size: 24px;
    font-weight: 300;
    color: #777;
    line-height: 32px
}

.main-search-inner {
    position: relative;
    display: block;
    top: 50%;
    transform: translate(0, -50%);
    padding-bottom: 30px
}

.main-search-inner select {
    border: 0;
    padding-top: 2px;
    padding-bottom: 0;
    height: 44px;
    box-shadow: none
}

.main-search-inner {
    position: absolute;
    display: block;
    bottom: 60px;
    top: 40vh;
    transform: none;
    padding-bottom: 0;
    z-index: 9999;
    margin: auto;
    width: 100%
}

.main-search-inner {
    transform: translate(0, -51%)
}

.container {
    margin-right: auto;
    margin-left: auto;
    padding-left: 15px;
    padding-right: 15px;
}

.video-container {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: -10;
    overflow: hidden
}

.main-search-container video {
    top: 50%;
    transform: translateY(-52%);
    left: 0;
    min-width: 100%;
    min-height: 100%;
    max-width: none;
    width: auto;
    height: auto;
    z-index: -10;
    background-size: cover;
    transition: 1s opacity;
    position: absolute
}

.main-search-container.dark-overlay:before {
    background: rgba(46, 47, 50, .55)
}

.main-search-container.dark-overlay h2 {
    color: #fff
}

.main-search-container.dark-overlay h4 {
    color: rgba(255, 255, 255, .7)
}


.main-search-input {
    margin-top: 50px;
    border-radius: 50px;
    width: 100%;
    background-color: #fff;
    box-shadow: 0 0 8px 0 rgba(0, 0, 0, .12);
    display: flex;
    flex-wrap: wrap;
    align-items: flex-start;
    padding: 9px;
    max-height: 68px
}

.main-search-input-item {
    flex: 1;
    border-right: 1px solid #e9e9e9;
    margin-top: 3px;
    position: relative;
    padding-left: 30px;
    padding-right: 30px
}

.main-search-input .main-search-input-item:nth-last-child(2) {
    border-right: 0;
    padding-left: 15px;
    padding-right: 15px
}

.main-search-input-item.location a {
    position: absolute;
    right: 20px;
    top: 50%;
    transform: translateY(-50%);
    color: #999;
    padding: 10px;
    background: #fff
}

.main-search-input-item.location a:hover {
    color: #66676b
}

.main-search-input input,
.main-search-input input:focus {
    font-size: 16px;
    border: 0;
    background: #fff;
    margin: 0;
    padding: 0;
    height: 44px;
    line-height: 44px;
    box-shadow: none
}

.main-search-input button.button {
    font-size: 18px;
    font-weight: 600;
    padding: 0 40px;
    margin-right: 1px;
    height: 50px;
    outline: 0
}

@media(max-width:1366px) {
    .main-search-container {
        height: 600px
    }
}

@media(max-width:991px) {
    .main-search-inner {
        top: 30vh;
    }

    .main-search-container {
        height: 580px
    }
    .main-search-input select,
    .main-search-input select:focus {
        font-size: 16px;
        border: 1px solid #e0e0e0;
        box-shadow: 0 1px 3px 0 rgba(0, 0, 0, .08);
        background: #fff;
        height: 55px;
        padding: 12px 18px;
        border-radius: 4px
    }

    .main-search-input {
        border-radius: 0;
        display: block;
        padding: 0;
        max-height: 100%;
        background-color: transparent;
        box-shadow: none
    }

    .main-search-input button.button {
        margin-top: 10px
    }
    .main-search-input-item {
        margin-bottom: 10px
    }
    .main-search-input-item,
    .main-search-input .main-search-input-item:nth-last-child(2) {
        padding: 0
    }
    .main-search-input-item .chosen-container-single .chosen-single,
    .main-search-input-item .chosen-single.chosen-container-active,
    .main-search-input input,
    .main-search-input input:focus {
        padding: 0 20px;
        height: 55px;
        line-height: 55px;
        border-radius: 4px;
        background: #fff;
        border: 1px solid #e0e0e0;
        box-shadow: 0 1px 3px 0 rgba(0, 0, 0, .08)
    }
    .main-search-input input,
    .main-search-input input:focus {
        padding: 0 20px
    }
    .main-search-input-item .chosen-container-single .chosen-single,
    .main-search-input-item .chosen-container-active.chosen-with-drop .chosen-single {
        border: 1px solid #e0e0e0;
        box-shadow: none;
        transition: 0s
    }
    .main-search-input-item .chosen-container-active.chosen-with-drop .chosen-single {
        border-bottom: 0
    }
    .main-search-input-item .chosen-container .chosen-drop {
        border: 1px solid #e0e0e0;
        border-top: 0;
        border-radius: 0 0 4px 4px;
        box-shadow: none
    }
    .main-search-input-item .chosen-single::after {
        display: none
    }
}


@media(max-width:767px) {
    .main-search-container h2 {
        font-size: 28px;
        font-weight: 400;
        line-height: 36px
    }
    .main-search-container h4 {
        font-size: 20px;
        font-weight: 400;
        line-height: 32px
    }
    .main-search-container:before {
        background: rgba(255, 255, 255, .92)
    }
}

.numbering{
    font-family: Roboto;
    font-style: normal;
    font-weight: bold;
    font-size: 64px;
    line-height: 75px;
    color: #C4C4C4;
}

.instruction {
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 21px;
}

.image-container {
    max-width: 350px;
    max-height: 400px;
}

.margin-right-20 {
    margin-right: 20px;
}

.padding-15 {
    padding: 15px;
}

.margin-15 {
    margin: 15px;
}

.margin-25 {
    margin: 25px;
}
