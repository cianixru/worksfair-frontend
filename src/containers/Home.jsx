import React, { Component, Fragment } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

import video from '../assets/main-search-video.mp4';
import background from '../assets/main-search-video-poster.jpg';
import SearchForm from '../forms/Public/SearchForm';
import Footer from '../components/Footer/Footer';

class HomeContainer extends Component {
  static propTypes = {
    history: PropTypes.object,
  }

  onSubmit = (data) => {
    if (!data.location) {
      data.location = 'Nigeria';
    }
    const { history } = this.props;
    history.push(`/search?query=${data.keywords}&location=${data.location}`);
  }

  render() {
    return (
      <Fragment>
        <Helmet>
          <title>Home - Worksfair</title>
        </Helmet>
        <div className="main-search-container dark-overlay">
          <div className="main-search-inner">
            <div className="container">
              <div className="columns">
                <div className="is-12">
                  <h2>Find Businesses and Services Near You.</h2>
                  <h4>Discover top-ranked businesses and services that are ready to serve your needs.</h4>

                  <SearchForm
                    onSubmit={this.onSubmit}
                  />
                </div>
              </div>
            </div>

          </div>

          <div className="video-container">
            <video poster={background} loop autoPlay muted>
              <source src={video} type="video/mp4" />
            </video>
          </div>
        </div>
        <Footer />
      </Fragment>
    );
  }
}

export default withRouter(HomeContainer);
