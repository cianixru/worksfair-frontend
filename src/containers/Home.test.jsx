import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import Home from './Home';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({
  auth: { currentUser: {} },
  webpage: { newWebpage: {} }
});

describe('Home.jsx', () => {
  test('Should display the Home Component', () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <Home />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
