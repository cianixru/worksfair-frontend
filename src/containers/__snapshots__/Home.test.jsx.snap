// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.jsx Should display the Home Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-search-container dark-overlay"
      >
        <div
          class="main-search-inner"
        >
          <div
            class="container"
          >
            <div
              class="columns"
            >
              <div
                class="is-12"
              >
                <h2>
                  Find Businesses and Services Near You.
                </h2>
                <h4>
                  Discover top-ranked businesses and services that are ready to serve your needs.
                </h4>
                <form>
                  <div
                    class="main-search-input"
                  >
                    <div
                      class="main-search-input-item"
                    >
                      <input
                        class="input"
                        name="keywords"
                        placeholder="What do you need?"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="main-search-input-item location"
                    >
                      <input
                        class="input"
                        name="location"
                        placeholder="Which city or town?"
                        type="text"
                        value=""
                      />
                    </div>
                    <button
                      class="button is-info is-rounded"
                      disabled=""
                    >
                      Search
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div
          class="video-container"
        >
          <video
            autoplay=""
            loop=""
            poster="main-search-video-poster.jpg"
          >
            <source
              src="main-search-video.mp4"
              type="video/mp4"
            />
          </video>
        </div>
      </div>
      <footer
        class="footer "
      >
        <div
          class="content has-text-centered"
        >
          Copyright 
          2019
           Worksfair. All Rights Reserved.
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-search-container dark-overlay"
    >
      <div
        class="main-search-inner"
      >
        <div
          class="container"
        >
          <div
            class="columns"
          >
            <div
              class="is-12"
            >
              <h2>
                Find Businesses and Services Near You.
              </h2>
              <h4>
                Discover top-ranked businesses and services that are ready to serve your needs.
              </h4>
              <form>
                <div
                  class="main-search-input"
                >
                  <div
                    class="main-search-input-item"
                  >
                    <input
                      class="input"
                      name="keywords"
                      placeholder="What do you need?"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="main-search-input-item location"
                  >
                    <input
                      class="input"
                      name="location"
                      placeholder="Which city or town?"
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    class="button is-info is-rounded"
                    disabled=""
                  >
                    Search
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        class="video-container"
      >
        <video
          autoplay=""
          loop=""
          poster="main-search-video-poster.jpg"
        >
          <source
            src="main-search-video.mp4"
            type="video/mp4"
          />
        </video>
      </div>
    </div>
    <footer
      class="footer "
    >
      <div
        class="content has-text-centered"
      >
        Copyright 
        2019
         Worksfair. All Rights Reserved.
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
